src/I18n/DateTime.php

```diff
@@ -126,11 +128,14 @@ class DateTime extends Chronos implements I18nDateTimeInterface, Stringable
      *
      * @param \DateTimeInterface|string|int|null $time Fixed or relative time
      * @param \DateTimeZone|string|null $tz The timezone for the instance
+     * @psalm-immutable
      */
     public function __construct(DateTimeInterface|string|int|null $time = null, DateTimeZone|string|null $tz = null)
     {
         if ($time instanceof DateTimeInterface) {
+            /** @psalm-suppress ImpureMethodCall */
```

Comment:
This implicit pure stuff will get annoying.

Reply:
Yeah, the error doesn't make sense to me for methods which return types other than the class itself.

Reply:
Can we disable purity checks entirely?

Reply:
Might be, I can check later.